subsys/bluetooth/controller/ll_sw/ull_sync.c

```diff
@@ -358,6 +370,65 @@ void ull_sync_release(struct ll_sync_set *sync)
 	mem_release(sync, &sync_free);
 }
 
+void ull_sync_setup_addr_check(struct ll_scan_set *scan, uint8_t addr_type,
+			       uint8_t *addr, uint8_t rl_idx)
+{
+	if (0) {
+
+#if defined(CONFIG_BT_CTLR_SYNC_PERIODIC_ADV_LIST)
+	/* Check if Periodic Advertiser list to be used */
+	} else if (scan->per_scan.filter_policy) {
+		/* Check in Periodic Advertiser List */
+		if (ull_filter_ull_pl_addr_match(addr_type, addr)) {
+			/* Remember the address, to check with
+			 * SID in Sync Info
+			 */
+			scan->per_scan.adv_addr_type = addr_type;
+			(void)memcpy(scan->per_scan.adv_addr, addr,
+				     BDADDR_SIZE);
+
+			/* Address matched */
+			scan->per_scan.state = LL_SYNC_STATE_ADDR_MATCH;
+
+#if defined(CONFIG_BT_CTLR_PRIVACY)
+		} else if (ull_filter_ull_pl_listed(rl_idx, &addr_type,
+						    scan->per_scan.adv_addr)) {
+			/* Remember the address, to check with
+			 * SID in Sync Info
+			 */
+			scan->per_scan.adv_addr_type = addr_type;
+
+			/* Address matched */
+			scan->per_scan.state = LL_SYNC_STATE_ADDR_MATCH;
+
+#else /* !CONFIG_BT_CTLR_PRIVACY */
+			ARG_UNUSED(addr_type);
+#endif /* !CONFIG_BT_CTLR_PRIVACY */
+
+		}
+#endif /* CONFIG_BT_CTLR_SYNC_PERIODIC_ADV_LIST */
+
+	/* Check with explicitly supplied address */
+	} else if ((addr_type == scan->per_scan.adv_addr_type) &&
+		   !memcmp(addr, scan->per_scan.adv_addr, BDADDR_SIZE)) {
+		/* Address matched */
+		scan->per_scan.state = LL_SYNC_STATE_ADDR_MATCH;
+	}
+}
+
+bool ull_sync_setup_sid_match(struct ll_scan_set *scan, uint8_t sid)
+{
+	return (scan->per_scan.state == LL_SYNC_STATE_ADDR_MATCH) &&
+		(0 ||
+#if defined(CONFIG_BT_CTLR_SYNC_PERIODIC_ADV_LIST)
+		 (scan->per_scan.filter_policy &&
+		  ull_filter_ull_pl_match(scan->per_scan.adv_addr_type,
+					  scan->per_scan.adv_addr, sid)) ||
+#endif /* CONFIG_BT_CTLR_SYNC_PERIODIC_ADV_LIST */
```

Comment:
Can't we do 
```suggestion
		(IS_ENABLED(CONFIG_BT_CTLR_SYNC_PERIODIC_ADV_LIST)
		 (scan->per_scan.filter_policy &&
		  ull_filter_ull_pl_match(scan->per_scan.adv_addr_type,
					  scan->per_scan.adv_addr, sid)) ||
```
here instead?

Reply:
done